src/common/imageio_module.c

```diff
@@ -437,6 +437,89 @@ void dt_imageio_insert_storage(dt_imageio_module_storage_t *storage)
   DT_DEBUG_CONTROL_SIGNAL_RAISE(darktable.signals, DT_SIGNAL_IMAGEIO_STORAGE_CHANGE);
 }
 
+void dt_imageio_scale_splitup(double *x, double *y)
+{
+  double _x, _y;
+  gchar *scale_str = dt_conf_get_string("plugins/lighttable/export/resizing_factor");
+
+  char sep[4] = "";
+  snprintf( sep, 4, "%g", (double) 3/2);
+  int i = -1;
+  while(scale_str[++i]) {
+      if ((scale_str[i] == '.') || (scale_str[i] == ',')) scale_str[i] = sep[1];
+  }
+
+  gchar *pdiv = strchr(scale_str, '/');
+
+  if (pdiv == NULL) {
+    _x = atof(scale_str);
+    _y = 1;
+  } else if (pdiv-scale_str == 0) {
+    _x = 1;
+    _y = atof(pdiv + 1);
+  } else {
+    _x = atof(scale_str);
+    _y = atof(pdiv+1);
+  }
+
+  if (_x == 0.0) _x = 1.0;
+  if (_y == 0.0) _y = 1.0;
+
+  *x = _x;
+  *y = _y;
+
+  free(scale_str);
+}
+
+
+void dt_imageio_scale_optim()
```

Comment:
can  you add some comments, not sure I understand all this business:

Reply:
This is where the string is cleared for better readability: from "1./2.0" is made "1/2".